### ISSUE TITLE
Could not determine the dependencies of task

### ISSUE THREAD

**Issue by PssbleTrngle** [modrinth/minotaur#28]
**Describe the bug**    

I have a multi-loader setup, publishing to modrinth on build.
It previously worked  (in May), but now I always get the following error:

<details>
<summary>
Crash Log
</summary>

```
Downloading https://services.gradle.org/distributions/gradle-7.3-bin.zip
...........10%...........20%...........30%...........40%...........50%...........60%...........70%...........80%...........90%...........100%
Welcome to Gradle 7.3!
Here are the highlights of this release:
 - Easily declare new test suites in Java projects
 - Support for Java 17
 - Support for Scala 3
For more details see https://docs.gradle.org/7.3/release-notes.html
To honour the JVM settings for this build a single-use Daemon process will be forked. See https://docs.gradle.org/7.3/userguide/gradle_daemon.html#sec:disabling_the_daemon.
Daemon will be stopped at the end of the build 
> Configure project :Common
SpongePowered Vanilla 'GRADLE' Toolset Version '0.2.1-SNAPSHOT'
Targeting Minecraft '1.18.2' on a 'JOINED' platform
> Configure project :Fabric
Fabric Loom: 0.10.66
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Using of the official minecraft mappings is at your own risk!
Please make sure to read and understand the following license:
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
# (c) 2020 Microsoft Corporation. These mappings are provided "as-is" and you bear the risk of using them. You may copy and use the mappings for development purposes, but you may not redistribute the mappings complete and unmodified. Microsoft makes no warranties, express or implied, with respect to the mappings provided here.  Use and modification of this document or the source code (in any form) of Minecraft: Java Edition is governed by the Minecraft End User License Agreement available at https://account.mojang.com/documents/minecraft_eula.
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
:remapping minecraft (TinyRemapper, official -> named)
:remapping minecraft (TinyRemapper, official -> intermediary)
Processing file: minecraft-project-@Fabric-mapped.jar
Processing file: minecraft-project-@Fabric-mapped.jar
:remapping 46 mods (TinyRemapper, intermediary -> named)
> Configure project :Forge
[MixinGradle] Skipping eclipse integration, extension not found
WARNING: This project is configured to use the official obfuscation mappings provided by Mojang. These mapping fall under their associated license, you should be fully aware of this license. For the latest license text, refer below, or the reference copy here: https://github.com/MinecraftForge/MCPConfig/blob/master/Mojang.md, You can hide this warning by running the `hideOfficialWarningUntilChanged` task
WARNING: (c) 2020 Microsoft Corporation. These mappings are provided "as-is" and you bear the risk of using them. You may copy and use the mappings for development purposes, but you may not redistribute the mappings complete and unmodified. Microsoft makes no warranties, express or implied, with respect to the mappings provided here.  Use and modification of this document or the source code (in any form) of Minecraft: Java Edition is governed by the Minecraft End User License Agreement available at https://account.mojang.com/documents/minecraft_eula.
Setting up MCP environment
Initializing steps
Executing steps
 > Running 'downloadManifest'
 > Running 'downloadJson'
 > Running 'downloadClient'
 > Running 'downloadServer'
 > Running 'extractServer'
 > Running 'downloadClientMappings'
 > Running 'mergeMappings'
 > Running 'stripClient'
 > Running 'stripServer'
 > Running 'merge'
 > Running 'listLibraries'
 > Running 'rename'
Stopping at requested step: /home/runner/.gradle/caches/forge_gradle/mcp_repo/de/oceanlabs/mcp/mcp_config/1.18.2-20220228.144236/joined/rename/output.jar
[08:13:33] [main/INFO]: Writing debug log file accesstransform.log
[08:13:33] [main/INFO]: Access Transformer processor running version 8.0.7+8.0.7+master.43473d43
[08:13:33] [main/INFO]: Command line arguments [--inJar, /home/runner/.gradle/caches/forge_gradle/minecraft_user_repo/net/minecraftforge/forge/1.18.2-40.0.22/forge-1.18.2-40.0.22-injected.jar, --outJar, /home/runner/.gradle/caches/forge_gradle/minecraft_user_repo/net/minecraftforge/forge/1.18.2-40.0.22_mapped_official_1.18.2/forge-1.18.2-40.0.22_mapped_official_1.18.2.jar, --logFile, accesstransform.log, --atFile, /home/runner/work/Skygrid/Skygrid/Forge/build/_atJar_4/parent_at.cfg]
[08:13:33] [main/INFO]: Reading from /home/runner/.gradle/caches/forge_gradle/minecraft_user_repo/net/minecraftforge/forge/1.18.2-40.0.22/forge-1.18.2-40.0.22-injected.jar
[08:13:33] [main/INFO]: Writing to /home/runner/.gradle/caches/forge_gradle/minecraft_user_repo/net/minecraftforge/forge/1.18.2-40.0.22_mapped_official_1.18.2/forge-1.18.2-40.0.22_mapped_official_1.18.2.jar
[08:13:33] [main/INFO]: Transformer file /home/runner/work/Skygrid/Skygrid/Forge/build/_atJar_4/parent_at.cfg
[08:13:33] [main/WARN]: Found existing output jar /home/runner/.gradle/caches/forge_gradle/minecraft_user_repo/net/minecraftforge/forge/1.18.2-40.0.22_mapped_official_1.18.2/forge-1.18.2-40.0.22_mapped_official_1.18.2.jar, overwriting
Creating SRG -> MCP TSRG
[08:13:38] [main/INFO]: JAR transformation complete /home/runner/.gradle/caches/forge_gradle/minecraft_user_repo/net/minecraftforge/forge/1.18.2-40.0.22_mapped_official_1.18.2/forge-1.18.2-40.0.22_mapped_official_1.18.2.jar
FAILURE: Build failed with an exception.
* What went wrong:
Could not determine the dependencies of task ':Fabric:modrinth'.
> Cannot convert /home/runner/work/Skygrid/Skygrid/Fabric/build/libs/skygrid-fabric-3.0.5.jar to a task.
  The following types/formats are supported:
    - A String or CharSequence task name or path
    - A Task instance
    - A TaskReference instance
    - A Buildable instance
    - A TaskDependency instance
    - A Provider that represents a task output
    - A Provider instance that returns any of these types
    - A Closure instance that returns any of these types
    - A Callable instance that returns any of these types
    - An Iterable, Collection, Map or array instance that contains any of these types
* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 3m [15](https://github.com/PssbleTrngle/Skygrid/runs/6932588163?check_suite_focus=true#step:8:16)s
Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.
You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.
See https://docs.gradle.org/7.3/userguide/command_line_interface.html#sec:command_line_warnings
Error: Process completed with exit code 1.
```
</details

**To Reproduce**    

My setup is [here](https://github.com/PssbleTrngle/Skygrid).
The plugin is applied & used in the `Forge/build.gradle` and `Fabric/build.gradle`

**Environment Info:**    
 - Gradle Version: 7.3
 - Minotaur Version: 2.3.1
 - API: production

**Comment by The-Fireplace** [comment]
I've noticed this as well, though I'm just now making the jump from Minotaur 1.1.0 to 2.X. My guess is the value given to `uploadFile` is being treated as a task even though previous versions allowed (and current documentation indicates) that a File instance is acceptable.
After plugging in specific versions, it seems Minotaur 2.2.1 introduced the issue, since I experience this problem with it (and newer), and don't with version 2.2.0.

**Comment by triphora** [comment]
Fixed in 835e60e and v2.3.2
